### DISCUSSION TITLE
Progress event listener not triggered

### DISCUSSION THREAD

**Discussion by walid-chekkouri** [google/model-viewer#4638] [Q&A] [answered]
I'm trying to add an event lsitner for the progress event in the model viewer but it doesn't call it when the progress bar appears!
I'm working in React :In this example the  console.log("HELLO"); is never called! 

Also, the load event works fine though on mobile but is never triggered on my windows PC (maybe it's another issue)!
I'm using the 3.3.0 version.
 useEffect(() => {

    if (!loaded) {
      const modelViewer = document.querySelector('#viewer');
      if (modelViewer) {

     
        const onProgress = (event) => {
          console.log("HELLO");
   
        };
        modelViewer.addEventListener('progress', onProgress);

        modelViewer.addEventListener("load", () => {

          console.log("model-viewer can activate AR: " + modelViewer.canActivateAR);
          setCanActivateAR(modelViewer.canActivateAR);

        });

        setLoaded(true);

      } else {
        console.log("no model viewer found");
      }

    }

  }, []);
  
  My model viewer looks something like this:
         <model-viewer
                      id="viewer"
                      style={{ width: '100%', height: '100%', borderRadius: '15px 50px', '--progress-bar-color': 'blue', '--progress-bar-height': '25px' }}

**Comment by elalish** [comment, marked as answer]
If you're serving locally, there's a chance it's loading so fast that it's finished by the time your script runs. You can check this by leaving `src` blank, and setting `src` only after you've added the event listener.

**Reply by walid-chekkouri** [reply]
Much thanks, that worked :)
